From 6a1e089a4c2b5e9df94947d591c26246602edbef Mon Sep 17 00:00:00 2001
From: Arindam Biswas <quic_arinbisw@quicinc.com>
Date: Fri, 14 Feb 2025 17:30:19 +0530
Subject: [PATCH] v4l2: Add support for V4L2_PIX_FMT_QC08C format

Upstream-Status: Inappropriate [Qualcomm specific]

Signed-off-by: Arindam Biswas <quic_arinbisw@quicinc.com>
---
 sys/v4l2/ext/videodev2.h | 1 +
 sys/v4l2/gstv4l2object.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/sys/v4l2/ext/videodev2.h b/sys/v4l2/ext/videodev2.h
index 2ceb1ca..e3b4ab3 100644
--- a/sys/v4l2/ext/videodev2.h
+++ b/sys/v4l2/ext/videodev2.h
@@ -775,6 +775,7 @@ struct v4l2_pix_format {
 #define V4L2_PIX_FMT_QC08C    v4l2_fourcc('Q', '0', '8', 'C') /* Qualcomm 8-bit compressed */
 #define V4L2_PIX_FMT_QC10C    v4l2_fourcc('Q', '1', '0', 'C') /* Qualcomm 10-bit compressed */
 #define V4L2_PIX_FMT_AJPG     v4l2_fourcc('A', 'J', 'P', 'G') /* Aspeed JPEG */
+#define V4L2_PIX_FMT_QC08C    v4l2_fourcc('Q', '0', '8', 'C') /* Qualcomm 8-bit compressed */
 
 /* 10bit raw packed, 32 bytes for every 25 pixels, last LSB 6 bits unused */
 #define V4L2_PIX_FMT_IPU3_SBGGR10	v4l2_fourcc('i', 'p', '3', 'b') /* IPU3 packed 10-bit BGGR bayer */
diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 288ff74..684eec8 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -191,6 +191,7 @@ static GstV4L2FormatDesc gst_v4l2_formats[] = {
   {MAP_FMT (NV24, NV24),                        KNOWN_DRM_MAP,                      GST_V4L2_RAW},
   {MAP_FMT (NV42, UNKNOWN),                     MAP_DRM (NV42, LINEAR),             GST_V4L2_RAW},
   {MAP_FMT (MM21, NV12_16L32S),                 KNOWN_DRM_MAP,                      GST_V4L2_RAW},
+  {MAP_FMT (QC08C, NV12_Q08C),                  KNOWN_DRM_MAP,                      GST_V4L2_RAW},
   {MAP_FMT (P010, P010_10LE),                   KNOWN_DRM_MAP,                      GST_V4L2_RAW},
 
   /* Bayer formats - see http://www.siliconimaging.com/RGB%20Bayer.htm */
-- 
2.25.1

